### ISSUE TITLE
Pam's HC compat issue

### ISSUE THREAD

**Issue by gallifreyhanyou** [AgriCraft/AgriCraft#1256]
## Quick description
flipped the switch to disable vanilla farming. certain newer pam's crops were suddenly unplantable on crop sticks, and could ONLY be planted in the vanilla style...completely opposite of what that button is supposed to do.

## Mods and versions used
* Minecraft: 1.12.2 Forge 14.23.5.2847
* AgriCraft: 2.12.0-1.12.0-a6
* InfinityLib: 1.12.0
* Pam's HarvestCraft: 1.12.2zf

## Steps to replicate the issue

1. Loaded up new world with vanilla planting disabled
2. Placed crops on farmland
3. new pam's crops like flax and amaranth will not allow planting on crops but will allow on normal vanilla farmland
4. this does not occur with old crops like tomato, turnip, etc. they get the message "vanilla planting disabled"
full list of reverse behavior crops: giga pickle, kale, agave, arrowroot, amaranth, cassava, chickpeas, elderberry, flax, green grape, huckleberry, jicama, Jute, kenaf, kohlrabi, lentil, millet, mulberry, quinoa, sisal, taro, tomatillo, juniper berry. 

## Your environment

* How are you playing?
  * Operating System: windows 10
  * Launcher: Java
  * Modpack: agricraft/pam's HC
* What kind of server:
  * [X] Singleplayer
  * [ ] Multiplayer
* Game Modes:
  * [X] Survival
  * [X] Creative
* Save File:
  * [X] Brand new save
  * [ ] Only an existing save

**Comment by OceanSnipe17** [comment]
Hey @gallifreyhanyou @InfinityRaider 

I have encountered the same issue and created the jsons needed for the missing plants and mutations. I copied them from the existing jsons and updated the code to the proper plants.

I did try to use similar jsons like for sisal i used cotton and for millet I used rye. etc.

I have tested all of the plants and some of the mutations. All the mutations show up in JEI properly though.

I also tried to use plants that did not mutate into anything else while still making sense for the mutations. Example: Millet is wheat and oats and Agave is sisal and cactus fruit.

Feel free to use/edit these as needed. just copy and paste them into the agricraft\json\defaults\mod_harvestcraft folder in the appropriate mutations and plants folder.

Message me if you have any questions.

I am using: 
AgriCraft-2.12.0-1.12.0-a6.jar
Pam's HarvestCraft 1.12.2zg.jar
forge-14.23.5.2847
MC Version 1.12.2

[Agricraft Corrections for Pams HC.zip](https://github.com/AgriCraft/AgriCraft/files/4961870/Agricraft.Corrections.for.Pams.HC.zip)
